Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,216 @@
+
+.. code-block:: php
+
+    // this is discouraged, obtain TSFE from request attribute 'frontend.controller'
+    // see next example
+    $cObj = $GLOBALS['TSFE']->cObj;
+
+.. code-block:: php
+
+    $frontendController = $request->getAttribute('frontend.controller');
+    $cObj = $frontendController->cObj;
+
+In the case of a non Extbase plugin via setter injection:
+
+.. code-block:: php
+
+    public function setContentObjectRenderer(ContentObjectRenderer $cObj): void
+    {
+        $this->cObj = $cObj;
+    }
+
+.. _tsfe_pageId:
+
+Access current page id
+----------------------
+
+Access the current page id:
+
+.. code-block:: php
+
+    // this is discouraged
+    $GLOBALS['TSFE']->id
+
+Can be done using the :ref:`'routing' <typo3-request-attribute-routing>`
+request attribute:
+
+.. code-block:: php
+
+    $pageArguments = $request->getAttribute('routing');
+    $pageId = $pageArguments->getPageId();
+
+.. _tsfe_language:
+
+Access language settings
+------------------------
+
+In order to get current language settings, such as the current language id,
+obtain :php:`\TYPO3\CMS\Core\Site\Entity\SiteLanguage` object from the
+:ref:`request attribute <request-attributes>` 'language':
+
+.. code-block:: php
+
+    // TYPO3\CMS\Core\Site\Entity\SiteLanguage object.
+    $language = $request->getAttribute('language');
+    $languageId = $language->getLanguageId();
+
+
+If the request is not available, accessing language settings
+can be done using the :ref:`language aspect <context_api_aspects_language>`.
+
+Get the language of the current page as integer:
+
+.. code-block:: php
+
+    $languageId = (int) $context->getPropertyFromAspect('language', 'id');
+
+.. _tsfe_frontendUser
+
+Access frontend user information
+--------------------------------
+
+Accessing information about Frontend users. Accessing
+:php:`$GLOBALS['TSFE']->fe_user` directly is discouraged.
+
+.. code-block:: php
+
+    // not recommended, use alternatives if possible
+    // TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication
+    $feUser = $GLOBALS['TSFE']->fe_user;
+
+Using :ref:`request-attributes`:
+
+.. code-block:: php
+
+    // TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication
+    $frontendUser = $request->getAttribute('frontend.user');
+
+Some information via frontend and backend users con be obtained via the
+:ref:`user aspect <context_api_aspects_user>`. For example:
+
+.. code-block:: php
+
+    // return whether a frontend user is logged in
+    $context->getPropertyFromAspect('frontend.user', 'isLoggedIn');
+
+.. _tsfe_baseURL:
+
+Get current base URL
+--------------------
+
+To get the base URL of the current site (or other site configuration), use site
+configuration:
+
+.. code-block:: php
+
+    // TYPO3\CMS\Core\Site\Entity\Site
+    $site = $request->getAttribute('site');
+    // array
+    $siteConfiguration = $site->getConfiguration();
+    $baseUrl = $siteConfiguration['base'];
+
+.. _tsfe_siteByPageId:
+
+Get site by page id
```

Comment:
```suggestion
Get site by page ID
```

I am also missing the discouraged usage of TSFE.

Reply:
Not sure why you capitalize id here.  I think both can we used or is this a convention?